data privacy: update ref app link and personal data annotations (#488)
cap-js/docs: guides/data-privacy/annotations.md

## Reference App Sample { #annotated-model }

> In the remainder of this guide, we use the [Incidents Management reference sample app](https://github.com/cap-js/incidents-app) as the base to add data privacy and audit logging to.

Reviewer: ```suggestion
In the remainder of this guide, we use the [Incidents Management reference sample app](https://github.com/cap-js/incidents-app) with some data model extensions (see below) as the base to add data privacy and audit logging to.
```

<img src="./assets/Incidents-App.drawio.svg" style="zoom:111%;" />

::: details Data Model Extensions…

In order to showcase sensitive data and data subject details, we need to extend the data model a bit as follows:

```cds [db/extensions.cds]
using from '@capire/incidents';

> using {
>   cuid,
>   managed
> } from '@sap/cds/common';

Reviewer: ```suggestion
using { cuid, managed } from '@sap/cds/common';
```

entity sap.capire.incidents.Addresses : cuid, managed {
  customer      : Association to sap.capire.incidents.Customers;
  city          : String;
  postCode      : String;
  streetAddress : String;
}

extend sap.capire.incidents.Customers with {
  creditCardNo : String(16) @assert.format: '^[1-9]\d{15}$';
  addresses    : Composition of many sap.capire.incidents.Addresses
                   on addresses.customer = $self;
};
```

:::

So, let's annotate the data model to identify personal data.
In essence, in all our entities we search for elements which carry personal data, such as person names, birth dates, etc., and tag them accordingly.
All found entities are classified as either *Data Subjects*, *Subject Details* or *Related Data Objects*.

Following the [best practice of separation of concerns](../domain-modeling#separation-of-concerns), we annotate our domain model in a separate file *srv/data-privacy.cds*, which we add to our project and fill it with the following content:

::: code-group

```cds [srv/data-privacy.cds]

> using {sap.capire.incidents as my} from '../db/extensions';

Reviewer: ```suggestion
using { sap.capire.incidents as my } from '../db/extensions';
```

::: code-group

```cds [srv/data-privacy.cds]

> using { sap.capire.incidents as db } from './processor-service';

Reviewer: ```suggestion
using { sap.capire.incidents as db } from '../db/schema';
```
this guide is written as if you'd clone incidents-app and work in that repo directly

Reviewer: but there are no addresses in incidents-app... we'll need to think of a solution

Reviewer: @mnkiefer let's discuss this in the next sync
